shelley/chain-and-ledger/executable-spec/src/Cardano/Ledger/Core.hs

```diff
@@ -0,0 +1,76 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+-- | This module defines core type families which we know to vary from era to
+-- era.
+--
+-- Families in this module should be indexed on era.
+--
+-- It is intended for qualified import:
+-- > import qualified Cardano.Ledger.Core as Core
+module Cardano.Ledger.Core
+  ( -- * Compactible
+    Compactible (..),
+    Compact (..),
+    ValType,
+    Value,
+  )
+where
+
+import Cardano.Binary (FromCBOR (..), ToCBOR (..))
+import Cardano.Prelude (NFData, NoUnexpectedThunks (..))
+import Data.Kind (Type)
+import Data.Typeable (Typeable)
+
+class
+  ( Compactible (Value era),
+    Eq (Value era),
+    FromCBOR (CompactForm (Value era)),
+    FromCBOR (Value era),
+    NFData (Value era),
+    NoUnexpectedThunks (Value era),
+    Show (Value era),
+    ToCBOR (CompactForm (Value era)),
+    ToCBOR (Value era),
+    Typeable (Value era)
+  ) =>
+  ValType era
```

Comment:
We should make this a type alias.

Also a different name? `EraValueIsVal`?

Also tempted to think this should live in the Shelley era rather than the "general" code